tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/util/LiferayWorkspaceUtil.java

```diff
@@ -356,4 +361,114 @@ public static IPath getHomeLocation( IProject project )
         return getHomeLocation( project.getLocation().toOSString() );
     }
 
+    public static File getWorkspaceDir( File dir )
+    {
+        return findParentFile( dir, new String[] { _SETTINGS_GRADLE_FILE_NAME, _GRADLE_PROPERTIES_FILE_NAME }, true );
+    }
+
+    public static String loadConfiguredHomeDir( String location )
+    {
+        return getLiferayWorkspaceGradleProperty( location, "liferay.workspace.home.dir", "bundles" );
+    }
+
+    public static String read( File file ) throws IOException
+    {
+        return new String( Files.readAllBytes( file.toPath() ) );
+    }
+
+    public static File findParentFile( File dir, String[] fileNames, boolean checkParents )
+    {
+
+        if( dir == null )
+        {
+            return null;
+        }
+
+        for( String fileName : fileNames )
+        {
+            File file = new File( dir, fileName );
+
+            if( file.exists() )
+            {
+                return dir;
+            }
+        }
+
+        if( checkParents )
+        {
+            return findParentFile( dir.getParentFile(), fileNames, checkParents );
+        }
+
+        return null;
+    }
+
+    public static String[] getLiferayWorkspaceProjectWarsDirs( final String workspaceLocation )
+    {
+        String[] retval = null;
+
+        if( workspaceLocation != null )
+        {
+            String val = getLiferayWorkspaceGradleProperty( workspaceLocation, "liferay.workspace.wars.dir", "wars" );
+
+            retval = val.split( "," );
+        }
+
+        return retval;
+    }
+
+    public static String getLiferayWorkspaceGradleProperty( String projectLocation, String key, String defaultValue )
+    {
+        File gradleProperties = new File( projectLocation, "gradle.properties" );
+
+        String retVal = null;
+
+        if( gradleProperties.exists() )
+        {
+            Properties properties = PropertiesUtil.loadProperties( gradleProperties );
+
+            retVal = properties.getProperty( key, defaultValue );
+        }
+
+        return retVal;
+    }
+
+    public static boolean isWorkspace( File dir )
```

Comment:
we already have other methods that check if it is workspace.  Why did you need to write another one?